File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/context/roleplay/GameRolePlayCharacterInformations.scala





















// Generated on 05/03/2013 10:35:04
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait GameRolePlayCharacterInformations extends GameRolePlayHumanoidInformations {
  override val typeId = 36

  protected trait Type extends GameRolePlayHumanoidInformations.Type with Serializable {
    override val typeId = GameRolePlayCharacterInformations.typeId

    val alignmentInfos: ActorAlignmentInformations.TypeImpl

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      alignmentInfos.serialize(buf)
    }

  }

  protected def create(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, name: String, humanoidInfo: HumanInformations.TypeImpl, accountId: Int, alignmentInfos: ActorAlignmentInformations.TypeImpl): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val contextualId = buf.readInt()
    val look = EntityLook.deserialize(buf)
    val disposition = types.get[EntityDispositionInformations.TypeImpl](buf.readShort());
    val name = buf.readString()
    val humanoidInfo = types.get[HumanInformations.TypeImpl](buf.readShort());
    val accountId = buf.readInt()
    val alignmentInfos = ActorAlignmentInformations.deserialize(buf)

    create(contextualId, look, disposition, name, humanoidInfo, accountId, alignmentInfos)
  }

}

object GameRolePlayCharacterInformations extends GameRolePlayCharacterInformations {
  case class TypeImpl(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, name: String, humanoidInfo: HumanInformations.TypeImpl, accountId: Int, alignmentInfos: ActorAlignmentInformations.TypeImpl) extends Type
  override def create(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, name: String, humanoidInfo: HumanInformations.TypeImpl, accountId: Int, alignmentInfos: ActorAlignmentInformations.TypeImpl) = TypeImpl(contextualId, look, disposition, name, humanoidInfo, accountId, alignmentInfos)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, name: String, humanoidInfo: HumanInformations.TypeImpl, accountId: Int, alignmentInfos: ActorAlignmentInformations.TypeImpl) = create(contextualId, look, disposition, name, humanoidInfo, accountId, alignmentInfos)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/GameRolePlayCharacterInformations.java


// Generated on 05/08/2013 19:38:01
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameRolePlayCharacterInformations extends GameRolePlayHumanoidInformations {
    public static final short TYPE_ID = 36;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public ActorAlignmentInformations alignmentInfos;
    
    public GameRolePlayCharacterInformations() { }
    
    public GameRolePlayCharacterInformations(int contextualId, EntityLook look, EntityDispositionInformations disposition, String name, HumanInformations humanoidInfo, int accountId, ActorAlignmentInformations alignmentInfos) {
        super(contextualId, look, disposition, name, humanoidInfo, accountId);
        this.alignmentInfos = alignmentInfos;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        alignmentInfos.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        alignmentInfos = new ActorAlignmentInformations();
        alignmentInfos.deserialize(buf);
    }
    
}
